### DISCUSSION TITLE
How to use the translate pattern with variable

### DISCUSSION THREAD

**Discussion by esseti** [danielmiessler/Fabric#1136] [Q&A] [answered]
Hello,
i've seen that the tranlate pattern has vartiables for the Language, such as `and perfectly in <Language> as possible.`
this is the pattern https://raw.githubusercontent.com/danielmiessler/fabric/refs/heads/main/patterns/translate/system.md

but how do I pass the variable there? should i use the `-v` How?

**Comment by jamescardona11** [comment, marked as answer]
Just pass the language at the end of the pattern.

![Screenshot 2024-11-14 at 1 29 26 PM](https://github.com/user-attachments/assets/e1321613-96e6-4e1d-bb0c-fc91d49b8b9b)

**Reply by esseti** [reply]
Great! how can instead use variables in patterns? the <> is the way to define them?
how should one provide different variables in input afterwards?
is there an example? i've some ideas for patterns that may accept variables

Thaks.

**Reply by bledburn** [reply]
If you want to pass variable with the fabric command, using the -v switch, whatever the variable name is within the pattern must be used in the -v switch using a key:value pair.


For instance, within the `translate` pattern, the variable name is lang_code, as identified in the pattern itself by `{{lang_code}}` and so the variable can be passed from the command line to the pattern with: 

`fabric -sp translate -v=lang_code:"ja-jp" < text_to_translate.txt

or the more well known:

cat text_to_translate.txt | fabric -sp translate -v=lang_code:"ja-jp"

I hope that makes sense. For a visual walk-thru, there is a a video here: https://www.youtube.com/watch?v=OlP9Pt5AxBE&t=344s

**Comment by esseti** [comment]
Coming back here, it seems that something odd is happening with that prompt. When I input "ita," it translates to Japanese instead of Italian. Additionally, sometimes the output is strange; the translation appears inline rather than as a standalone translation. Is anyone else experiencing this?

**Reply by esseti** [reply]
![image](https://github.com/user-attachments/assets/5cdb3a9e-f7a8-49d0-817f-83a5625d3f51)
Proof 🇯🇵 🇮🇹

**Reply by esseti** [reply]
esp is now korean (?)

![image](https://github.com/user-attachments/assets/560ed157-89df-4be0-a325-0c0e81b8ba24)

**Reply by bledburn** [reply]
To truly know and figure out what is happening, you need to run that full command, followed by `--dry-run` and then post the output back here. That will show us exactly what the fabric tool is sending in the payload to the LLM, along with the LLM (model) and model parameters being used. The root cause can then start to be traced.
